### PR TITLE
Mark expired certificates during certificate dump

### PR DESCRIPTION
**Pull request by dadrian**

### CODE REVIEW

File: src/admin.cc [hunk 1]
```diff
@@ -491,11 +491,26 @@ class AdminServiceImpl final : public zsearch::AdminService::Service {
...
+            // `not_valid_after` are actually set.
+            if (it->second.certificate().not_valid_before() &&
+                it->second.certificate().not_valid_after()) {
+                std::time_t now = std::time(nullptr);
+                bool expired =
+                        !certificate_valid_at(it->second.certificate(), now);
+                if (expired != it->second.certificate().expired()) {
+                    it->second.mutable_certificate()->set_expired(expired);
+                    AnonymousResult res = cert_store.put(it->second);
+                    delta_handler->handle_delta(res);
```
Comment (zakird) [inline src/admin.cc]: We also need to set `valid` and `trusted_path` based on what's in `was_valid` and `was_trusted_path`.

**Comment by dadrian** [comment]
@zakird Can you give this a second pass?

File: src/admin.cc [hunk 2]
```diff
@@ -392,11 +392,44 @@ class AdminServiceImpl final : public zsearch::AdminService::Service {
...
+        std::time_t now = std::time(nullptr);
         for (auto it = cert_store.seek_start_of_shard(shard_id);
              it.valid() && cert_store.shard_for(it->first) == shard_id; ++it) {
             if (max_records && ++count > max_records) {
                 return;
             }
+            // Only update expiration if `not_valid_before` and
+            // `not_valid_after` are actually set.
+            if (it->second.certificate().not_valid_before() &&
+                it->second.certificate().not_valid_after()) {
```
Comment (zakird) [inline src/admin.cc]: can you fix this indent?

File: src/admin.cc [hunk 3]
```diff
@@ -392,11 +392,44 @@ class AdminServiceImpl final : public zsearch::AdminService::Service {
                                     std::mutex& out_mtx,
                                     uint32_t max_records,
                                     std::atomic<std::uint32_t>& count) {
+        std::unique_ptr<DeltaHandler> delta_handler =
+                m_delta_ctx->new_certificate_delta_handler();
+        std::unique_ptr<DeltaHandler> to_process_handler =
+                m_delta_ctx->new_certificates_to_process_delta_handler();
+        std::time_t now = std::time(nullptr);
         for (auto it = cert_store.seek_start_of_shard(shard_id);
              it.valid() && cert_store.shard_for(it->first) == shard_id; ++it) {
```
Comment (zakird) [inline src/admin.cc]: indent is off.

File: src/admin.cc (hunk 2 above)
Comment (dadrian) [inline src/admin.cc]: what's wrong with the indent?

File: src/admin.cc (hunk 3 above)
Comment (dadrian) [inline src/admin.cc]: once again, not seeing anything wrong with the indent.

**Pull request merged**